pydron.py

```diff
@@ -0,0 +1,156 @@
+#!/venv/bin/python
+
+import shlex
+import subprocess
+import sys
+import time
+
+import requests
+from six.moves.BaseHTTPServer import BaseHTTPRequestHandler, HTTPServer
+
+
+def _print(out):
+    """
+    Print to stderr, so it shows up in the sytest logs when TAP is being used.
+    """
+    sys.stderr.write(str(out) + "\n")
+    sys.stderr.flush()
+
+
+def wait_for_start(process, url):
+    """
+    Wait for the start of Synapse by polling its HTTP endpoint.
+    """
+    tries = 10
+    while tries != 0:
+        tries -= 1
+
+        try:
+            r = requests.get("http://" + url, timeout=5)
+            if r.status_code:
+                # We don't care what the status code is, just that it's
+                # responding.
+                return
+        except Exception as e:
+            time.sleep(2)
```

Comment:
I think we should poll more frequently than every 2s TBH, otherwise its going to take an age to start up. Something like one every 200ms is probably fine.